nomad/structs/services.go

```diff
@@ -257,6 +267,14 @@ func (sc *ServiceCheck) validate() error {
 		}
 	}
 
+	if sc.SuccessBeforePassing < 0 {
```

Comment:
The [Consul docs](https://www.consul.io/docs/agent/checks.html#success-failures-before-passing-critical
) say:
> This feature is available for HTTP, TCP, gRPC, Docker & Monitor checks. 

so we should probably validate that the check isn't a script check here?

Reply:
good idea, added!